fairscale/nn/pipe/multiprocess_pipe.py

```diff
@@ -259,32 +256,29 @@ def __init__(
                 f" {len(self.balance)})"
             )
 
+        if isinstance(module, nn.Sequential):
+            local_partitions = split_module(module, self.balance)
+            self._skip_layout = inspect_skip_layout(local_partitions)
+        else:
+            self._skip_layout = SkipLayout(len(module), {})  # FIXME(tom)
+
         rank = self.group.rank()
+        self.final_stage = rank == len(self.balance) - 1
         if rank >= len(self.balance):
             warnings.warn("More ranks than partitions, some ranks unused")
```

Comment:
do we really need to support this case with code coverage, unit test etc or should just assert fail?

Reply:
Good point. Could would be simpler if we just remove this and error.